src/readcalls.c

```diff
@@ -240,20 +203,22 @@ int readcalls(void) {
 	pfxnumcntidx = -1;
 	pxnr = 0;
 
-	/* get bandindex */
-	bandindex = log_get_band(inputbuffer);
+	qso = parse_qso(inputbuffer);
+	bandindex = qso->bandindex;
 
 	/* get the country number, not known at this point */
-	g_strlcpy(presentcall, inputbuffer + 29, 14);
-	tmpptr = strchr(presentcall, ' ');
-	if (tmpptr)
-	    *tmpptr = '\0';
+	countrynr = getctydata(qso->call);
+
+	/*  lookup worked stations, add if new */
+	int station = lookup_or_add_worked(qso->call);
+	update_worked(station, qso);
 
-	countrynr = getctydata(presentcall);
 
 	if (continentlist_only) {
-	    if (!is_in_continentlist(continent)) {
+	    if (!is_in_continentlist(dxcc_by_index(countrynr)->continent)) {
 		qsos_per_band[bandindex]++;
+		free_qso(qso);
+		qso = NULL;
```

Comment:
is this needed? or just a preventive measure in case `qso` would be used later on?

Reply:
Not really. But it helps to catch 'use after free' errors during development.